torchao/prototype/qat/nvfp4.py

```diff
@@ -23,47 +22,159 @@ class NVFP4FakeQuantizeConfig(FakeQuantizeConfigBase):
     Args:
         use_per_tensor_scale (bool): Whether to use two-level per-tensor fp32 scaling
             after the initial fp8 (e4m3) block-wise scaling (default True)
+        use_swizzled_scales (bool): Whether scales are stored in swizzled (blocked) format
+        use_triton_kernel (bool): Whether to use triton kernels during fake quantization
     """
 
     use_per_tensor_scale: bool = True
+    use_swizzled_scales: bool = False
+    use_triton_kernel: bool = False
 
 
-class NVFP4FakeQuantizer(FakeQuantizerBase):
+class _NVFP4FakeQuantizedLinearForward(torch.autograd.Function):
```

Comment:
seems like this isn't `FakeQuantizedLinear`, it's actually doing quantized forward and only fake quantizing in the backward?  The name should correctly describe what this is doing.